modules/rabbitmq/src/main/java/org/testcontainers/containers/RabbitMQContainer.java

```diff
@@ -0,0 +1,358 @@
+package org.testcontainers.containers;
+
+import com.github.dockerjava.api.command.InspectContainerResponse;
+import org.jetbrains.annotations.NotNull;
+import org.json.JSONObject;
+import org.testcontainers.containers.wait.strategy.LogMessageWaitStrategy;
+import org.testcontainers.containers.wait.strategy.WaitAllStrategy;
+import org.testcontainers.containers.wait.strategy.WaitStrategy;
+import org.testcontainers.utility.MountableFile;
+
+import java.io.IOException;
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Stream;
+
+import static java.lang.String.join;
+import static java.util.Arrays.asList;
+import static java.util.Collections.emptyList;
+import static java.util.stream.Collectors.toSet;
+
+/**
+ * Testcontainer for RabbitMQ.
+ *
+ * @author Martin Greber
+ */
+public class RabbitMQContainer<S extends RabbitMQContainer<S>> extends GenericContainer<S> {
+
+    /**
+     * The image defaults to the official RabbitmQ image: <a href="https://hub.docker.com/_/rabbitmq/">RabbitMQ</a>.
+     */
+    private static final String DEFAULT_IMAGE_NAME = "rabbitmq";
+    private static final String DEFAULT_TAG = "3.7-management-alpine";
+
+    private static final int DEFAULT_AMQP_PORT = 5672;
+    private static final int DEFAULT_AMQPS_PORT = 5671;
+    private static final int DEFAULT_HTTPS_PORT = 15671;
+    private static final int DEFAULT_HTTP_PORT = 15672;
+
+    private String adminPassword = "guest";
+    private String adminUsername = "guest";
+    private List<String> plugins = emptyList();
+    private List<List<String>> values = new ArrayList<>();
+
+    /**
+     * Creates a Testcontainer using the official Neo4j docker image.
+     */
+    public RabbitMQContainer() {
+        this(DEFAULT_TAG);
+    }
+
+    /**
+     * Creates a Testcontainer using a specific docker image.
+     *
+     * @param tag The docker image to use.
+     */
+    public RabbitMQContainer(String tag) {
+        super(DEFAULT_IMAGE_NAME + ":" + tag);
+
+        WaitStrategy waitForStartup = new LogMessageWaitStrategy()
+                .withRegEx(String.format(".*Server startup complete.*"));
+
+        this.waitStrategy = new WaitAllStrategy()
```

Comment:
Why is it WaitAll, if it has only one check?

Also, is it really necessarily to wait for the log, does it open the port before the container is ready?

Reply:
Yes, It starts listening to the port before all initialisation (like queue creation) is done. This is because at the moment the RabbitMQContainer uses the rabbitmqadmin command to setup everything, and it runs after RabbitMQ starts listening.

Another possibility is to set up all configuration in a json file, and pass that to RabbitMQ. It would load the configuration on statup. However, there are some other difficulties with this approach. I might try it later though.

Reply:
If you're doing the initialization in `containerIsStarted`, the startup will not finish unless `containerIsStarted` is executed. 
It means that after `container.start()`, it will run the wait strategy (open port) and apply the admin commands in `containerIsStarted`.
Would it be enough?

Reply:
Just waiting for the management port to open _might_ be enough, but I'm not sure. I think for now the safest is to wait for the 'Server startup complete' log message. Theoretically, if the management port is up we should be able to run management commands. Though I don't think that will save much time.

Container log:
```
....
<0.554.0> Management plugin: HTTP (non-TLS) listener started on port 15672
<0.660.0> Statistics database started.
<0.8.0> Server startup complete; 3 plugins started.
...
```